mesonbuild/compilers/detect.py

```diff
@@ -648,6 +649,13 @@ def detect_cuda_compiler(env: 'Environment', for_machine: MachineChoice) -> Comp
         cpp_compiler = detect_cpp_compiler(env, for_machine)
         cls = CudaCompiler
         env.coredata.add_lang_args(cls.language, cls, for_machine, env)
+        key = OptionKey('cuda_link_args', machine=for_machine)
+        if key in env.options:
+            # To fix LDFLAGS issue
+            val = env.options[key]
+            flags: T.List[str] = []
+            flags.extend(val)
+            env.coredata.set_options({key: cls.to_host_flags_base(flags, Phase.LINKER)})
```

Comment:
What exactly is the flags.extend(val) doing here?

Reply:
If you're simply concerned that mypy should agree that `env.options[key]` is a list of str, and it is always expected to be that way, then maybe you should do this instead:

```
flags = env.options[key]
assert isinstance(key, list), 'for mypy'
```

Reply:
Thanks, this assertion is enough to pass mypy. I have changed accordingly.